src/libraries/Common/src/System/Net/Security/Unix/SafeDeleteContext.cs

```diff
@@ -29,6 +31,16 @@ protected SafeDeleteContext(SafeFreeCredentials credential)
             _credential.DangerousAddRef(ref ignore);
         }
 
+        protected void SetCredentials(SafeFreeCredentials credential)
```

Comment:
Do we really needs this? The Unix PAL does not use the credentials, right? I know this is how Windows work.

Reply:
I was not completely sure, about that, there is some refcount going on on the creds, and it also felt weird to keep a reference to credentials that are not actually used anymore.